loopchain/blockchain/invoke_result.py

```diff
@@ -0,0 +1,238 @@
+            "prevBlockValidators": prev_block_validators,
+            "prevBlockVotes": prev_block_votes
+        }
+
+    def _serialize_txs(self) -> List[Union[dict]]:
+        if not self._transactions:
+            return []
+
+        transactions = []
+        for tx in self._transactions.values():
+            tx_serializer = TransactionSerializer.new(tx.version, tx.type(), self._tx_versioner)
+            transaction = {
+                "method": "icx_sendTransaction",
+                "params": tx_serializer.to_full_data(tx)
+            }
+            transactions.append(transaction)
+
+        return transactions
+
+    def _serialize_prev_votes_except_leader(self) -> List[List[Union[str]]]:
+        """Serialize all prev votes except leader's.
+
+        Return example:
+            [[voter_id, vote_result], ...]
+        """
+        prev_block_votes = []
+        for vote in self._prev_votes:
+            # Remove Leader's Vote
+            if vote.voter_id.hex_hx() == self._prev_peer_id:
+                continue
+
+            if vote.is_none():
+                vote_result = "0x2"
+            elif vote.is_lazy():
+                vote_result = "0x0"
+            elif vote.is_real():
+                vote_result = "0x1"
+            else:
+                raise RuntimeError("Failed to parse vote result.")
+
+            prev_block_votes.append([vote.voter_id.hex_hx(), vote_result])
+
+        return prev_block_votes
+
+    def _extract_validators(self, prev_block_votes: List[List[str]]) -> List[str]:
+        """Get all validators except leader."""
+        if prev_block_votes:
+            validators, _vote_results = zip(*prev_block_votes)
+            return [validator for validator in validators]
+        else:
+            return []
+
+
+class InvokeData(Message):
+    def __init__(self,
+                 epoch_num: int,
+                 round_num: int,
+                 added_transactions: Dict[str, dict],
+                 reps_hash: Hash32,
+                 next_prep: dict = None):
+        """Represents a return value of Invoke Message, received from ICON-Service.
+
+        :param epoch_num: Current epoch number
+        :param round_num: Current round number
+        :param added_transactions: Txs added by ICON-Service
+        :param reps_hash: Current reps hash
+        :param next_prep: Information about next reps.
+
+        Additional params:
+            next_preps: A group of next preps
+            next_preps_hash: reps_hash of `next_preps`
+            changed_reason: A reason why the prep list has been changed
+        """
+        self._id: bytes = f"{epoch_num}_{round_num}".encode()
+        self._epoch_num: int = epoch_num
+        self._round_num: int = round_num
+        self._added_transactions: Dict[str, dict] = added_transactions
+        self._reps_hash: Hash32 = reps_hash
+
+        self._next_preps: Optional[list] = None
+        self._next_preps_hash: Hash32 = Hash32.new()
```

Comment:
In block 1.0, _next_preps_hash must be filled.(in this case with reps_hash) Block 1.0 does not use special blocks. That rule is only for 0.3 blocks.